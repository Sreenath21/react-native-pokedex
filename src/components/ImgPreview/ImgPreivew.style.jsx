import {StyleSheet} from 'react-native';

export default StyleSheet.create({
  thumbnailContainer: {
    flex: 1,
    borderColor: 'black',
    borderWidth: 1,
    borderStyle: 'dashed',
    borderRadius: 10,
    overflow: 'hidden',
  },
  gradient: {
    flex: 1,
    borderRadius: 10,
  },
  image: {
    overflow: 'hidden',
    flex: 1,
    width: 150,
    alignSelf: 'center',
    justifyContent: 'center',
  },
});
