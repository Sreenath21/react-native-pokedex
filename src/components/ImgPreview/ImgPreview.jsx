import React, {useContext} from 'react';
import {View, Image} from 'react-native';
import LinearGradient from 'react-native-linear-gradient';

import {PokemonContext} from '../../context/PokemonContext';
import {getGradient} from '../../utils/helplerFunctions';
import styles from './ImgPreivew.style';

const ImgPreview = () => {
  const {pokemon} = useContext(PokemonContext);

  return (
    <View style={styles.thumbnailContainer}>
      <LinearGradient style={styles.gradient} colors={getGradient(pokemon)}>
        <Image
          style={styles.image}
          source={{
            uri: `https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/${pokemon.name}.png`,
          }}
          resizeMode="contain"
        />
      </LinearGradient>
    </View>
  );
};

export default ImgPreview;
