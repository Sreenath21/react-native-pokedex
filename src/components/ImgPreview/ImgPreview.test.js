import React from 'react';
import ImgPreview from './ImgPreview';
import {render} from '@testing-library/react-native';
import PokemonContextProvider from '../../context/PokemonContext';

it('ImgPreview renders correctly', () => {
  const tree = render(<ImgPreview />, {wrapper: PokemonContextProvider});
  expect(tree).toMatchSnapshot();
});
