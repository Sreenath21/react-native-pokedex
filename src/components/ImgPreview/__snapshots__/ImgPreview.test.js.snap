// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ImgPreview renders correctly 1`] = `
<View
  style={
    {
      "borderColor": "black",
      "borderRadius": 10,
      "borderStyle": "dashed",
      "borderWidth": 1,
      "flex": 1,
      "overflow": "hidden",
    }
  }
>
  <BVLinearGradient
    colors={
      [
        undefined,
      ]
    }
    endPoint={
      {
        "x": 0.5,
        "y": 1,
      }
    }
    locations={null}
    startPoint={
      {
        "x": 0.5,
        "y": 0,
      }
    }
    style={
      {
        "borderRadius": 10,
        "flex": 1,
      }
    }
  >
    <Image
      resizeMode="contain"
      source={
        {
          "uri": "https://img.pokemondb.net/sprites/omega-ruby-alpha-sapphire/dex/normal/undefined.png",
        }
      }
      style={
        {
          "alignSelf": "center",
          "flex": 1,
          "justifyContent": "center",
          "overflow": "hidden",
          "width": 150,
        }
      }
    />
  </BVLinearGradient>
</View>
`;
